pycroft/lib/task.py

```diff
@@ -192,10 +198,22 @@ def get_active_tasks_by_type(type):
 
 
 @with_transaction
-def cancel_task(task, processor):
+def cancel_task(task: Task, processor: User):
     if task.status != TaskStatus.OPEN:
         raise ValueError("Cannot cancel a task that is not open")
 
     log_task_event("Cancelled task {}.".format(task.id), processor, task)
 
     task.status = TaskStatus.CANCELLED
+
+
+def manually_execute_task(task: Task, processor: User):
+    if task.status != TaskStatus.OPEN:
+        raise ValueError("Cannot execute a task that is not open")
+
+    get_task_implementation(task).execute(task)
```

Comment:
Catching errors here could be a good idea, because when an exception occurs, the user only sees a 500 error message.

Reply:
The question would be which errors to catch. A `except Exception:` would not be advisable. We want to have programming errors be fatal and reported.
However, if it's an error of the form „does not live in that room“ or sth, we will probably want to catch that.

But what does that mean to thhe lifecycle of the task? Does it go to `TaskState.FAILED`?

Reply:
I don't think that it is an good idea to reimplement the task execution logic here again. So we should either

- Do not change the state to `FAILED` but just display an error message instead

or

- Don't execute the task here directly, but instead trigger that the task should be executed by celery. For example by setting the due date to the current timestamp and calling `pycroft.task.execute_scheduled_tasks.delay()`.

Reply:
I think this is a good time to introduce a `PycroftException` for `lib` errors.

Reply:
See #493

Reply:
Resolved in f34fea5261b30ba55a6e5c5a5a00775a195427b3.